swarm/src/lib.rs

```diff
@@ -64,6 +64,7 @@ mod upgrade;
 pub mod behaviour;
 pub mod dial_opts;
 pub mod dummy;
+pub mod executor;
```

Comment:
```suggestion
mod executor;
```

Lets do this instead but re-export the trait.